streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractReadOnlyDecorator.java

```diff
@@ -68,6 +70,8 @@ public void close() {
     static StateStore getReadOnlyStore(final StateStore global) {
         if (global instanceof TimestampedKeyValueStore) {
             return new TimestampedKeyValueStoreReadOnlyDecorator<>((TimestampedKeyValueStore<?, ?>) global);
+        } else if (global instanceof VersionedKeyValueStore) {
```

Comment:
This addition, and the analogous one in `AbstractReadWriteDecorator`, is necessary so that `ProcessorContextImpl#getStateStore()` can properly return versioned stores.